### PR TITLE
Address various ways to misuse Assert.Fail

### PR DESCRIPTION
**Pull request by mayerj**

### CODE REVIEW

File: Philips.CodeAnalysis.MsTestAnalyzers/AssertFailAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		private const string Description = @"";
+		private const string Category = Categories.Maintainability;
+
+		private static DiagnosticDescriptor Rule = new DiagnosticDescriptor(Helper.ToDiagnosticId(DiagnosticIds.AssertFail), Title, MessageFormat, Category, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
```
Comment (bcollamore) [inline Philips.CodeAnalysis.MsTestAnalyzers/AssertFailAnalyzer.cs:30]: We probably don't want these flags, correct? ie., don't run on generated code, which is presumably not very relevant for an mstest.

File: Philips.CodeAnalysis.MsTestAnalyzers/Philips.CodeAnalysis.MsTestAnalyzers.csproj [hunk 2]
```diff
@@ -24,7 +24,7 @@
     <PackageTags>CSharp MsTest Roslyn CodeAnalysis analyzers Philips</PackageTags>
     <NoPackageAnalysis>true</NoPackageAnalysis>
     <DevelopmentDependency>true</DevelopmentDependency>
-    <Version>1.1.1</Version>
+    <Version>1.1.2</Version>
     <AssemblyVersion>1.0.1.0</AssemblyVersion>
     <FileVersion>1.1.1</FileVersion>
```
Comment (bcollamore) [inline Philips.CodeAnalysis.MsTestAnalyzers/Philips.CodeAnalysis.MsTestAnalyzers.csproj:29]: Increment FileVersion

**Review by bcollamore — APPROVED** [review]

File: Philips.CodeAnalysis.MsTestAnalyzers/AssertFailAnalyzer.cs (hunk 1 above)
Comment (mayerj) [reply Philips.CodeAnalysis.MsTestAnalyzers/AssertFailAnalyzer.cs]: This is actually consistent with the rest of the mstest analyzers, but yes, we don't care about it in generated code.  Removed.

File: Philips.CodeAnalysis.MsTestAnalyzers/Philips.CodeAnalysis.MsTestAnalyzers.csproj (hunk 2 above)
Comment (mayerj) [reply Philips.CodeAnalysis.MsTestAnalyzers/Philips.CodeAnalysis.MsTestAnalyzers.csproj]: Done.

**Review by bcollamore — APPROVED** [review]

**Pull request merged**